arrow/arrow.py

```diff
@@ -563,7 +573,11 @@ def span(
             floor = self.__class__(*values, tzinfo=self.tzinfo)  # type: ignore
 
             if frame_absolute == "week":
-                floor = floor.shift(days=-(self.isoweekday() - 1))
+                delta = 0
+                # if weekday is greater than self.isoweekday() go back one week by setting delta = 7
+                if week_start > self.isoweekday():
+                    delta = 7
+                floor = floor.shift(days=-(self.isoweekday() - week_start) - delta)
```

Comment:
Could a modulo be used here rather than a subtraction to wrap around?

Reply:
Hey @jadchaar ,
I made attempts using modulo, but without success. My assumption is that provided date must still be in the week that is spaned. Modulo might wrap the week but I need to forceably move back a week, hence -7 days.
Example:
``` # 2021-03-18 is thursday, 5 is isoweekday of Friday```
```arrow.get("2021-03-18").span('week', week_start=5)```
```>>> (<Arrow [2021-03-12T00:00:00+00:00]>, <Arrow [2021-03-18T23:59:59.999999+00:00]>)```

Does this make sense to you?